server/src/refactor.ts

```diff
@@ -6,22 +6,26 @@
  */
 
 import * as lsp from 'vscode-languageserver';
-import tsp from 'typescript/lib/protocol';
+import * as tsp from 'typescript/lib/protocol';
 import { Commands } from './commands';
 
-export function provideRefactors(response: tsp.GetApplicableRefactorsResponse | undefined, result: (lsp.Command | lsp.CodeAction)[], args: tsp.FileRangeRequestArgs): void {
+export function provideRefactors(
+    response: tsp.GetApplicableRefactorsResponse | undefined,
+    args: tsp.FileRangeRequestArgs
+): Array<lsp.CodeAction> {
     if (!response || !response.body) {
-        return;
+        return [];
     }
-    for (const info of response.body) {
+    return response.body.reduce<Array<lsp.CodeAction>>((actions, info) => {
         if (info.inlineable === false) {
-            result.push(asSelectRefactoring(info, args));
+            actions.push(asSelectRefactoring(info, args));
         } else {
-            for (const action of info.actions) {
-                result.push(asApplyRefactoring(action, info, args));
-            }
+            actions.push(...info.actions.map(action =>
+                asApplyRefactoring(action, info, args)
+            ));
         }
-    }
+        return actions;
+    }, []);
```

Comment:
Isn't this changed version equal to the previous version while being also less readable (IMO)?
Please revert if so because it was fine as is.

Reply:
The main difference is tho return an array, rather than mutate a param. I can convert it to not use `reduce` though.